roofit/roofitcore/src/RooDataSet.cxx

```diff
@@ -744,27 +744,33 @@ RooDataSet::RooDataSet(const char *name, const char *title, TTree *intree,
 
 
 ////////////////////////////////////////////////////////////////////////////////
-/// Constructor of a data set from (part of) an ROOT TTRee. The dimensions
-/// of the data set are defined by the 'vars' RooArgSet. For each dimension
+/// Constructor of a data set from (part of) a ROOT TTree.
+///
+/// \param[in] name Name of this dataset.
+/// \param[in] title Title for e.g. plotting.
+/// \param[in] ntuple Tree to be imported.
+/// \param[in] vars Defines the columns of the data set. For each dimension
 /// specified, the TTree must have a branch with the same name. For category
 /// branches, this branch should contain the numeric index value. Real dimensions
 /// can be constructed from either 'Double_t' or 'Float_t' tree branches. In the
 /// latter case, an automatic conversion is applied.
-///
-/// The 'cuts' string is an optional
-/// RooFormula expression and can be used to select the subset of the data points 
-/// in 'dset' to be copied. The cut expression can refer to any variable in the
-/// vars argset. For cuts involving variables other than those contained in
-/// the vars argset, such as intermediate formula objects, use the 
-/// equivalent constructor accepting RooFormulaVar reference as cut specification
-///
-
-RooDataSet::RooDataSet(const char *name, const char *title, TTree *intree, 
-		       const RooArgSet& vars, const char *selExpr, const char* wgtVarName) :
+/// \param[in] cuts Optional RooFormula expression to select the subset of the data points
+/// to be imported. The cut expression can refer to any variable in `vars`.
+/// \warning The expression only evaluates variables that are also in `vars`.
+/// Passing e.g.
+/// ```
+/// RooDataSet("data", "data", tree, RooArgSet(x), "x>y")
+/// ```
+/// Will load `x` from the tree, but leave `y` at an undefined value.
```

Comment:
not for this PR, but can this not be diagnosed?

Reply:
I thought about it, but not easily. The problem is that RooFit and TTree could both do the work, but Wouter decided to do the work on the RooFit side. At the end of the day, a RooFormulaVar uses a TFormula, which is getting compiled, and the compilation finishes fine. See this:
```
root [0] TFormula formula("formula", "x>y")
(TFormula &) Name: formula Title: x>y
root [2] formula.Eval(3)
(double) 1.0000000
root [3] formula.Eval(4)
(double) 1.0000000
root [4] formula.Eval(5)
(double) 1.0000000
root [5] formula.Eval(1)
(double) 1.0000000
root [6] formula.Eval(0)
(double) 0.0000000
root [7] formula.Eval(-1)
(double) 0.0000000
root [8] 
```

Reply:
what..is TFormula doing? xD

Reply:
This to be precise:
```
Double_t TFormula::Eval(Double_t x) const
{
   double * xxx = &x;
   return EvalPar(xxx, nullptr);
}
```
And that ends up in:
https://github.com/root-project/root/blob/74a7a81c8371863e40ac5d3eac2cf11093bbc92b/hist/hist/src/TFormula.cxx#L3329

Reply:
Here, you can pass any array length you want. :-)

Reply:
that's horrible...